src/Microsoft.AspNet.Mvc.Core/UnitTestability/MvcControllerUnitTestHelperCallback.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using Microsoft.AspNet.Http.Core.Collections;
+
+namespace Microsoft.AspNet.Mvc.Core
+{
+	public class MvcControllerUnitTestHelperCallback
+	{
+		public Func<CancellationToken> OnRquestAborted
+		{
+			get;
+			set;
+		}
+		public Func<FormCollection> OnRequestFormCollection
+		{
+			get;
+			set;
+		}
```

Comment:
Take a look at https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNet.Http.Core/DefaultHttpContext.cs

In general all of the `HttpContext` stuff is get/set properties. In general setting a property is much simpler to get your head around than using callbacks, so we should leverage that there the default is testable. If you find things in that space that don't work well let's loop in the appropriate folks and improve it.